### PR TITLE
cmd,daemon: add debug command for displaying the base policy

### PR DESCRIPTION
**Pull request by zyga**
The security team has requested that we add a command to inspect the base policy affecting
all interfaces. This has to be done in advance of any changes to the way the base policy is
assembled so that the system remains easy to audit.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: daemon/api.go [hunk 1]
```diff
@@ -2241,6 +2241,9 @@ func postDebug(c *Command, r *http.Request, user *auth.UserState) Response {
 		st.Lock()
 		defer st.Unlock()
 		ensureStateSoon(st)
+	case "get-base-declaration":
+		blob := asserts.Encode(asserts.BuiltinBaseDeclaration())
+		return SyncResponse(blob, nil)
```
Comment (pedronis) [inline daemon/api.go]: if you look in response.go  there's a specific helper to return assertions

Comment (pedronis) [inline daemon/api.go]: also you want assertstate.BaseDeclaration(st) , not using directly asserts, see ifacestate

Comment (zyga) [inline daemon/api.go]: Thanks I'll make the modifications

Comment (pedronis) [inline daemon/api.go]: don't know if you saw my comment on IRC; but I had forgotten about Client.Debug.... so using a SyncResponse is probably ok but we might some labeling/wrapping

map[string]interface{}{"base-declaration": blob}

File: cmd/snap/cmd_get_base_declaration_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		default:
+			c.Fatalf("expected to get 1 requests, now on %d", n+1)
+		}
+
+		n++
+	})
+	rest, err := snap.Parser().ParseArgs([]string{"debug", "get-base-declaration"})
+	c.Assert(err, check.IsNil)
+	c.Assert(rest, check.DeepEquals, []string{})
+	c.Check(s.Stdout(), check.Equals, "hello\n")
```
Comment (mvo5) [inline cmd/snap/cmd_get_base_declaration_test.go]: Sorry for being slow, but where does this "hello" come from? I see above that there is a  `"result": "aGVsbG8K"` but no "hello"?

While at it, I think I would prefer something slightly closer to the real output from the daemon, i.e. if someone looks at this test ideally he/she would get a feel what kind of data to expect from the daemon.

Comment (zyga) [inline cmd/snap/cmd_get_base_declaration_test.go]: The result is of type []byte which gets base64 encoded.

File: daemon/api.go (hunk 1 above)
Comment (zyga) [inline daemon/api.go]: Done

File: cmd/snap/cmd_get_base_declaration_test.go (hunk 2 above)
Comment (zyga) [inline cmd/snap/cmd_get_base_declaration_test.go]: I changed this to a regular string for the principle of least surprise.

**Pull request merged**